modules/configuration/src/main/java/org/apache/ignite/internal/configuration/notifications/ConfigurationNotificationUtils.java

```diff
@@ -0,0 +1,134 @@
+            DynamicConfiguration<InnerNode, ?> dynamicConfig,
+            String nodeName
+    ) {
+        return (DynamicConfiguration<InnerNode, ?>) dynamicConfig.members().get(nodeName);
+    }
+
+    /**
+     * Returns the named dynamic configuration of the child node.
+     *
+     * @param dynamicConfig Dynamic configuration.
+     * @param nodeName      Name of the child node.
+     * @return Named dynamic configuration of the child node or {@code null} if the child node does not exist.
+     */
+    static @Nullable NamedListConfiguration<?, InnerNode, ?> namedDynamicConfig(
+            DynamicConfiguration<InnerNode, ?> dynamicConfig,
+            String nodeName
+    ) {
+        return (NamedListConfiguration<?, InnerNode, ?>) dynamicConfig.members().get(nodeName);
+    }
+
+    /**
+     * Null-safe version of {@link ConfigurationNode#listeners()}.
+     *
+     * <p>Needed for working with "any" configuration properties, see this class' javadoc for details.
+     *
+     * @return Listeners of the given node or an empty list if it is {@code null}.
+     */
+    static <T> Collection<ConfigurationListener<T>> listeners(@Nullable ConfigurationNode<T> node) {
+        return node == null ? List.of() : node.listeners();
+    }
+
+    /**
+     * Null-safe version of {@link NamedListConfiguration#extendedListeners()}.
+     *
+     * <p>Needed for working with "any" configuration properties, see this class' javadoc for details.
+     *
+     * @return Listeners of the given node or an empty list if it is {@code null}.
+     */
+    static <T> Collection<ConfigurationNamedListListener<T>> extendedListeners(@Nullable NamedListConfiguration<?, T, ?> node) {
+        return node == null ? List.of() : node.extendedListeners();
+    }
+
+    /**
+     * Returns the dynamic configuration of the {@link NamedListConfiguration#any any} node.
+     *
+     * @param namedConfig Dynamic configuration.
+     * @return Dynamic configuration of the "any" node.
+     */
+    static @Nullable DynamicConfiguration<InnerNode, ?> any(@Nullable NamedListConfiguration<?, InnerNode, ?> namedConfig) {
+        return namedConfig == null ? null : (DynamicConfiguration<InnerNode, ?>) namedConfig.any();
+    }
+
+    /**
+     * Merge {@link NamedListConfiguration#any "any"} configurations.
+     *
+     * @param anyConfigs Current {@link NamedListConfiguration#any "any"} configurations.
+     * @param anyConfig  New {@link NamedListConfiguration#any "any"} configuration.
+     * @return Merged {@link NamedListConfiguration#any "any"} configurations.
+     */
+    static Collection<DynamicConfiguration<InnerNode, ?>> mergeAnyConfigs(
+            Collection<DynamicConfiguration<InnerNode, ?>> anyConfigs,
+            @Nullable DynamicConfiguration<InnerNode, ?> anyConfig
+    ) {
+        return Stream.concat(anyConfigs.stream(), Stream.of(anyConfig))
+                .filter(Objects::nonNull)
```

Comment:
How can they contain nulls? Is it possible?

Reply:
Yes, because of the polymorphic configuration.